src/components/Explore/state.ts

```diff
@@ -1,4 +1,26 @@
+import { SetStateAction } from 'jotai'
 import { atomWithStorage } from 'jotai/utils'
 
 export const favoritesAtom = atomWithStorage<string[]>('favorites', [])
 export const showFavoritesAtom = atomWithStorage<boolean>('showFavorites', false)
+
+/* handle favorite token logic */
+export const toggleFavoriteToken = ({
+  tokenAddress,
+  favoriteTokens,
+  updateFavoriteTokens,
+}: {
+  tokenAddress: string
+  favoriteTokens: string[]
+  updateFavoriteTokens: (update: SetStateAction<string[]>) => void
+}) => {
+  let updatedFavoriteTokens
```

Comment:
can't you use the atom in this function instead of passing stuff in?

Reply:
yes i thought that too but useAtom is a hook -- so then it would need to either be called in a react component or function (not super familiar with that part of React so could that just be useEffect or..?)

Reply:
make a hook called `useToggleFavorite`